### PR TITLE
#5648: fix permissions error when activating from extension console

### LINKED ISSUE #5648: Fix permissions error when activating from Extension Console on first run

### PR DESCRIPTION
**Pull request by twschiller**
## What does this PR do?

- Closes #5648 
- Wraps Extension Console activation wizard in a gate to ensure brick definitions are loaded locally prior to activation/re-activation
- Adds a `RequireBrickRegistry` gate that enforces brick definitions are loaded locally before rendering children
- Refactors duplicate URL logic into `useActivateUrl`
- Renamed `ActivateWizard` to `ActivateWizardCard` to clarify it's role in the UI

## Demo

- _Paste a screenshot or demo video here_

## Future Work

- Clean up more of useWizard and related components using the async state PR: https://github.com/pixiebrix/pixiebrix-extension/pull/5622

## Checklist

- [x] Add tests: Rainforest test
- [x] Designate a primary reviewer: @mnholtz

### CODE REVIEW

File: src/extensionConsole/pages/activateRecipe/ActivateBlueprintPage.tsx [hunk 1]
```diff
@@ -18,28 +18,23 @@
...
-  }>();
-
-  const reinstall =
-    new URLSearchParams(location.search).get("reinstall") === "1";
+  const { blueprintId, isReinstall } = useActivateUrl();
 
+  // Fetch the latest version of the blueprint, to ensure activate/re-activation is for the latest version
   const {
     data: remoteBlueprint,
     isLoading: fetchingBlueprint,
```
Comment (mnholtz) [inline src/extensionConsole/pages/activateRecipe/ActivateBlueprintPage.tsx:40]: nit (unrelated to feature): just realized, is there a reason we're not using RTK query here? this method isn't in `api.ts`

File: src/extensionConsole/pages/activateRecipe/useActivateUrl.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+/**
+ * Hook to parse information from an Extension Console activation URL, and redirect to the re-activation page as
+ * necessary.
+ */
+function useActivateUrl() {
+  const dispatch = useDispatch();
+  const location = useLocation();
+
+  // Could this be moved to useParams? Or is it on the main URL instead of the hash?
```
Comment (mnholtz) [inline src/extensionConsole/pages/activateRecipe/useActivateUrl.ts:41]: `useParams` actually doesn't refer to search query params, unfortunately: https://reactrouter.com/en/main/hooks/use-params

File: src/extensionConsole/pages/activateRecipe/ActivateBlueprintPage.tsx [hunk 3]
```diff
@@ -60,7 +55,19 @@ const ActivateBlueprintPage: React.FunctionComponent = () => {
 
   const body = useMemo(() => {
     if (remoteBlueprint?.config?.extensionPoints) {
-      return <ActivateWizard blueprint={recipeDefinition} />;
+      // Require that bricks have been fetched at least once before showing. Handles new installs where the bricks
+      // haven't been completely fetched yet.
+      // XXX: we might also want to enforce a full re-sync of the brick registry to ensure the latest brick
```
[the remark this answers is not in the document]
Comment (twschiller) [reply src/extensionConsole/pages/activateRecipe/ActivateBlueprintPage.tsx:60]: What I use for something where the behavior might not be quite right/we might consider changing in the future. But it's not a TODO

File: src/extensionConsole/pages/activateRecipe/useActivateUrl.ts (hunk 2 above)
Comment (mnholtz) [reply src/extensionConsole/pages/activateRecipe/useActivateUrl.ts:41]: Noticed that there's now a `useSearchParams`? Not sure when this was added (I've tried to find this hook before) https://reactrouter.com/en/main/hooks/use-search-params-rn

Comment (mnholtz) [reply src/extensionConsole/pages/activateRecipe/useActivateUrl.ts:41]: Maybe we can consider upgrading eventually

**Review by mnholtz — APPROVED** [review]

**Review by BLoe — APPROVED** [review]

File: src/extensionConsole/pages/activateRecipe/ActivateBlueprintPage.tsx (hunk 1 above)
Comment (twschiller) [reply src/extensionConsole/pages/activateRecipe/ActivateBlueprintPage.tsx:42]: I clarified the reason in the comment -- we need to make sure we're always getting the latest recipe definition

Comment (mnholtz) [reply src/extensionConsole/pages/activateRecipe/ActivateBlueprintPage.tsx:42]: I think you can force refetch: https://redux-toolkit.js.org/rtk-query/api/createApi#refetchonmountorargchange

**Pull request merged**